### PR TITLE
17: Routing II - TopBar

### PR DESCRIPTION
**Pull request by gabrielizalo**
# Summary

* 17: Routing II - TopBar
   Se adicionaron las siguientes páginas: Profile, Error 404, Offline.

# Trello Card

https://trello.com/c/uJVzXHFq/25-routing-ii-topbar

# Screenshot

screenshot_2018-11-20-09-53-19
screenshot_2018-11-20-09-54-11
screenshot_2018-11-20-09-54-26
gabos tica tac toe - login
gabos tica tac toe - profile
gabos tica tac toe - game
gabos tica tac toe - 404
gabos tica tac toe - offline

### CODE REVIEW

File: react/react-training/public/index.html [hunk 1]
```diff
@@ -17,6 +17,8 @@
     <link rel="mask-icon" href="%PUBLIC_URL%/safari-pinned-tab.svg" color="#333132">
     <meta name="msapplication-TileColor" content="#333132">
     <meta name="theme-color" content="#333132">
+
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
```
Comment (henryzarza) [inline react/react-training/public/index.html]: Move this to `application.scss`

File: react/react-training/src/app/components/AppRoutes/index.js [hunk 2]
```diff
@@ -2,21 +2,34 @@ import connect from 'react-redux/es/connect/connect';
...
   render() {
-    const { userIsLogged } = this.props;
+    if (OFFLINE_WE_ARE) {
+      return <Offline />;
+    }
+
+    // We are online...
+    let { userIsLogged } = this.props;
+    if (!userIsLogged) {
+      userIsLogged = localStorage.getItem('userIsLogged') === 'true';
```
Comment (henryzarza) [inline react/react-training/src/app/components/AppRoutes/index.js]: Isn't a good practice modify the props

File: react/react-training/src/app/screens/Dashboard/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import Game from './screens/Game';
+import Logout from './screens/Logout';
+import Profile from './screens/Profile';
+import TopBar from './screens/Game/components/TopBar';
+
+function Dashboard() {
+  return (
+    <Router>
+      <Fragment>
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/index.js:13]: This `Fragment` is required?

File: react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+import styles from './styles.scss';
+
+function TopBar() {
+  return (
+    <nav className={styles.bar}>
+      <ul className={styles.list}>
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js]: Isn't necessary this nesting? I recommend you modify the structure

File: react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        <li className={styles.item}>
+          <i className="fas fa-gamepad" />
+          <NavLink activeClassName={styles.active} className={styles.link} to="/game/">
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js]: Put the routes as constants

File: react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            Game
+          </NavLink>
+        </li>
+        <li className={styles.item}>
+          <i className="fas fa-user" />
+          <NavLink activeClassName={styles.active} className={styles.link} to="/profile/">
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js]: The same here

File: react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            Profile
+          </NavLink>
+        </li>
+        <li className={styles.item}>
+          <i className="fas fa-sign-out-alt" />
+          <NavLink activeClassName={styles.active} className={styles.link} to="/logout/">
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js]: And here

File: react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/styles.scss [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  padding: 2px 0 2px 0;
+}
+
+.item {
+  cursor: pointer;
+  font-size: $top-bar-size;
+  padding: 15px;
+}
+
+.item:hover, .item a:hover{
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/styles.scss]: Use the nesting of SASS

File: react/react-training/src/app/screens/Dashboard/screens/Logout/index.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import { actionsCreators as Actions } from '@redux/login/actions';
+
+class Logout extends Component {
+  componentDidMount() {
+    const { logout } = this.props;
+    logout();
+  }
+
+  render() {
+    return <Fragment />;
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Logout/index.js]: I think that isn't necessary this screen, please delete it

File: react/react-training/src/app/screens/Dashboard/screens/Logout/styles.scss [hunk 10]
```diff
@@ -0,0 +1,8 @@
+@import 'variables/colors';
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Logout/styles.scss]: And delete this

File: react/react-training/src/app/screens/Dashboard/screens/Profile/index.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import styles from './styles.scss';
+
+function Profile() {
+  return (
+    <div>
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Profile/index.js]: If you have only one element isn't necessary this

File: react/react-training/src/app/screens/Error404/index.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+import cry from './assets/404.png';
+import styles from './styles.scss';
+
+function Error404() {
+  return (
+    <div>
+      <h2 className={styles.title}>{ERROR_404}</h2>
+      <p className={styles.image}>
+        <img alt={ERROR_404} height={cry.height} src={cry} width={cry.width} />
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Error404/index.js]: Isn't a good practice to assign the properties `width` and `height` here, please move it to `styles` file

File: react/react-training/src/app/screens/Offline/index.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+import offline from './assets/offline.png';
+import styles from './styles.scss';
+
+function Offline() {
+  return (
+    <div>
+      <h2 className={styles.title}>{OFFLINE_MESSAGE}</h2>
+      <p className={styles.image}>
+        <img alt={OFFLINE_MESSAGE} height={offline.height} src={offline} width={offline.width} />
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Offline/index.js]: The same here

File: react/react-training/src/scss/styles.scss [hunk 14]
```diff
@@ -2,7 +2,11 @@
 @import 'variables/colors';
 
 body {
+  align-items: center;
   background-color: $mine-shaft;
   color: $white;
   font-family: $font-main;
+  justify-content: center;
+  margin: 0;
```
Comment (henryzarza) [inline react/react-training/src/scss/styles.scss]: This is necessary?

File: react/react-training/src/scss/styles.scss [hunk 15]
```diff
@@ -2,7 +2,11 @@
...
+  padding: 0;
```
Comment (henryzarza) [inline react/react-training/src/scss/styles.scss]: The same question here

File: react/react-training/src/scss/variables/_fonts.scss [hunk 16]
```diff
@@ -5,3 +5,7 @@ $font-main: 'Roboto', Helvetica, sans-serif;
 $title-size: 25px;
 $boxes-font-size: 15px;
 $button-font-size: 15px;
+
+$top-bar-size: 12px;
+
```
Comment (henryzarza) [inline react/react-training/src/scss/variables/_fonts.scss]: Delete this line

File: react/react-training/src/scss/variables/_fonts.scss [hunk 17]
```diff
@@ -5,3 +5,7 @@ $font-main: 'Roboto', Helvetica, sans-serif;
 $title-size: 25px;
 $boxes-font-size: 15px;
 $button-font-size: 15px;
+
```
Comment (henryzarza) [inline react/react-training/src/scss/variables/_fonts.scss]: Delete this line

**Review by henryzarza — CHANGES_REQUESTED** [review]

File: react/react-training/.env [hunk 18]
```diff
@@ -1 +1 @@
-REACT_APP_API_BASE_URL=http://localhost:3001/api/
+REACT_APP_API_BASE_URL=http://localhost:3002/api/
```
Comment (wfolini) [inline react/react-training/.env]: Remember to delete this file from repository

File: react/react-training/src/app/screens/Dashboard/index.js (hunk 3 above)
Comment (gabrielizalo) [reply react/react-training/src/app/screens/Dashboard/index.js:13]: @henryzarza yesss... Si no lo tengo me sale este error:
`A <Router> may have only one child element`

**Comment by gabrielizalo** [comment]
We are readyyyyyyy.. @WFolini make your move 👍

**Review by wfolini — CHANGES_REQUESTED** [review]

**Comment by gabrielizalo** [comment]
We are super ok 👍

**Review by wfolini — APPROVED** [review]

**Review by henryzarza — APPROVED** [review]

**Pull request merged**